src/atoms_calculators.jl

```diff
@@ -0,0 +1,106 @@
+#
+# Define AtomsCalculators interface for DFTK.
+#
+# This interface is inspired by the one used in Molly.jl, 
+# see https://github.com/JuliaMolSim/Molly.jl/blob/master/src/types.jl
+#
+# TODO: Decide how Hamiltonian terms should be updated upon change of the positions.
+# These terms are held inside the `basis`. Currently, the whole basis is rebuilt.
+# TODO: Find out if symmetries can be re-enabled (see issue on GH).
+
+struct DFTKState
+    scfres::NamedTuple
+end
+function construct_dummy_state(basis)
+    ρ = guess_density(basis)
+    ψ = nothing # Will get initialized by SCF.
+    return DFTKState((;ρ, ψ, basis))
+end
```

Comment:
Actually all we need should be `DFTKState(system::AbstractSystem, params::DFTKParameters)`. From this you should be able to get a decent dummy state and do the `prepare_basis`